wremnants/datasets/datagroups.py

```diff
@@ -57,20 +57,20 @@ def __init__(self, infile, combine=False, datasets=None):
                         "name": g_name,
                         "group": g_name,
                         "filepaths": group["dataset"]["filepaths"],
-                        "xsec": None
+                        "xsec": group["dataset"]["xsec"],
                         })
-            else:
-                self.datasets = {x.name : x for x in datasets}
 
-            logger.debug(f"Getting these datasets: {self.datasets.keys()}")
-
-            if self.results:
                 self.data = [x for x in self.datasets.values() if x.is_data]
                 if self.data:
                     self.lumi = sum([self.results[x.name]["lumi"] for x in self.data if x.name in self.results])
```

Comment:
Everything looks good to me, for convenience here I would add a logger.debug (or maybe even as info?) printing the integrated luminosity from the read out data